src/backend/joanie/tests/core/test_api_admin_contract_definitions.py

```diff
@@ -0,0 +1,160 @@
+
+        response = self.client.get(
+            f"/api/v1.0/admin/contract-definitions/{certification_definition.id}/"
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["id"], str(certification_definition.id))
+
+    def test_admin_api_contract_definition_create(self):
+        """
+        Staff user should be able to create a contract definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        data = {
+            "title": "Contract 001",
+            "template": "joanie.core.templates.contract-001",
+            "language": "fr-fr",
+        }
+
+        response = self.client.post(
+            "/api/v1.0/admin/contract-definitions/",
+            content_type="application/json",
+            data=data,
+        )
+
+        self.assertEqual(response.status_code, 201)
+        content = response.json()
+        self.assertIsNotNone(content["id"])
+        self.assertEqual(content["title"], "Contract 001")
+
+    def test_admin_api_contract_definition_update(self):
+        """
+        Staff user should be able to update a contract definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        certification_definition = factories.ContractDefinitionFactory(
+            title="Contract grade 1", name=enums.CONTRACT_NAME_CHOICES[0][0]
+        )
+        payload = {
+            "title": "Updated Contract grade 1",
+            "name": enums.CONTRACT_NAME_CHOICES[0][0],
+            "language": "fr-fr",
+        }
+
+        response = self.client.put(
+            f"/api/v1.0/admin/contract-definitions/{certification_definition.id}/",
+            content_type="application/json",
+            data=payload,
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["id"], str(certification_definition.id))
+        self.assertEqual(content["title"], "Updated Contract grade 1")
+        self.assertEqual(content["name"], "contract_definition")
+
+    def test_admin_api_contract_definition_partially_update(self):
+        """
+        Staff user should be able to partially update a certification_definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        certification_definition = factories.ContractDefinitionFactory(
+            name=enums.CONTRACT_DEFINITION
+        )
+
+        response = self.client.patch(
+            f"/api/v1.0/admin/contract-definitions/{certification_definition.id}/",
+            content_type="application/json",
+            data={"name": enums.CONTRACT_DEFINITION},
```

Comment:
You are not testing this is updated as it is the same value.
Why not updating the title instead ?

Reply:
Good point. Done!